### PR TITLE
Fix cosmos link

### LINKED ISSUE #291: Fix cosmos link
```
FILE: ./not-so-smart-contracts/cosmos/non_determinism/README.md
[✖] https://lev.pm/posts/2020-04-18-golang-map-randomness/

9 links checked.

ERROR: 1 dead links found!
[✖] https://lev.pm/posts/2020-04-18-golang-map-randomness/ → Status: 404
```

### PR DESCRIPTION
**Pull request by kshitiz305**
FILE: ./not-so-smart-contracts/cosmos/non_determinism/README.md [✖] https://lev.pm/posts/2020-04-18-golang-map-randomness/

Fixes #291

### CODE REVIEW

**Comment by kshitiz305** [comment]
HI @GrosQuildu request you to kindly review the changes

**Comment by montyly** [comment]
Thanks @kshitiz305 for the contribution. 

I am a bit unsure about this link, the page is full of ads. Maybe we can use an alternative? For example:

- https://go.dev/blog/maps 
- https://freshman.tech/snippets/go/iterate-over-map/

Any of those would probably be better

**Comment by kshitiz305** [comment]
@Messerblatt provided me with the link in the PR. Shall I update it? 

https://github.com/crytic/building-secure-contracts/issues/291#issuecomment-1532591546


@montyly Please do let me know if you plan to update the links

File: not-so-smart-contracts/cosmos/non_determinism/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Non-determinism in conensus-relevant code will cause the blockchain to halt.
 There are quite a few sources of non-determinism, some of which are specific to the Go language:
 
-- [`range` iterations over an unordered map or other operations involving unordered structures](https://lev.pm/posts/2020-04-18-golang-map-randomness/)
+- [`range` iterations over an unordered map or other operations involving unordered structures](https://www.delftstack.com/howto/go/how-to-iterate-over-an-entire-map-in-go/#iterate-over-all-keys-and-values-of-a-map-in-go)
```
Comment (GrosQuildu) [inline not-so-smart-contracts/cosmos/non_determinism/README.md:6]:
```suggestion
- [`range` iterations over an unordered map or other operations involving unordered structures](https://go.dev/blog/maps#iteration-order)
```

**Review by GrosQuildu — APPROVED** [review]

**Comment by montyly** [comment]
Thanks to @kshitiz305 and @GrosQuildu

**Pull request merged**